### DISCUSSION TITLE
Use kafka connector without kafka connect?

### DISCUSSION THREAD

**Discussion by dudicoco** [strimzi/strimzi-kafka-operator#9360] [Q&A]
Hi,

Is it possible to use the `KafkaConnector` custom resource without using the `KafkaConnect` resource?

It seems that currently the only option is to use the `strimzi.io/cluster` annotation within the connector resource, is it possible to somehow provide it with the endpoint of an existing kafka connect service insteade?

We are managing the kafka connect deployment without strimzi and would only like to use strimzi to create and configure the connectors via the API.

**Comment by scholzj** [comment]
No, that is not possibly. It has to be used with the KafkaConnect resource
and Strimzi managed Kafka Connect.

**Reply by dudicoco** [reply]
Thanks @scholzj.
Is there a reason for this limitation? Would it be possible to add support for an annotation which specifies the endpoint of a connect cluster instead?

**Reply by scholzj** [reply]
It has separate resources. Bu the connector reconciliation is handled by the same code as the Connect itself. So it cannot be easily separated / would require significant effort.

I guess the interesting question is why don't you just use Strimzi's Connect as well 😉.

**Reply by dudicoco** [reply]
@scholzj what would be the benefit of using strimzi's connect over a simple deployment resource?

**Reply by scholzj** [reply]
You get the operator to manage your Kafka Connect deployment and you will get to use the KafkaConnector resources for managing connectors? 😉

**Reply by dudicoco** [reply]
@scholzj the KafkaConnector resources part is obvious :)
I'm asking specifically about the kafka connect deployment, what benefit does the operator give me over a deployment which i can manage myself?

**Reply by scholzj** [reply]
Like with all Kubernetes operators, it is supposed to do at least some of the operations for you. It also gives you the possibility to run things in a declarative fashion and use tools such as GitOps - unless you already run your current Connect on Kubernetes. Probably less complicated with Connect than with the brokers ... but might still have some advantages. Whether it is worth it for you or not, that is always up to you.

**Reply by dudicoco** [reply]
what are some of the operations that the operator + kafka connect resource do for you that a normal deployment object does not?
